### PR TITLE
removes mocked functions in unit test instead sets requests expected …

### PR DESCRIPTION
**Pull request by TizianoPerrucci**
…responses

### Step 1: Link to Jira issue
https://stackstate.atlassian.net/browse/STAC-11444

### Step 3: Did you add / update tests for your changes in the right area?
- [x] Unit/Component test
- [ ] Integration test	

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py [hunk 1]
```diff
@@ -301,4 +299,4 @@ def get_dynatrace_json_response(self, endpoint, timeout=10):
                 return yaml.safe_load(resp.text)
```
Comment (vishalxebia) [inline dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py:299]: @TizianoPerrucci do we want to swap this as well with `json.load` as in tests you already converted to `json.load`?

Comment (TizianoPerrucci) [reply dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py:299]: @vishalxebia yes next ticket, another PR coming

File: dynatrace_topology/tests/test_dynatrace_topology.py [hunk 2]
```diff
@@ -55,24 +58,23 @@ def test_collect_empty_topology(self):
...
+    @requests_mock.Mocker()
+    def test_collect_processes(self, m):
         """
         Testing Dynatrace check should collect processes
         """
+        m.get("/api/v1/entity/infrastructure/hosts", text="[]")
+        m.get("/api/v1/entity/applications", text="[]")
+        m.get("/api/v1/entity/infrastructure/process-groups", text="[]")
+        m.get("/api/v1/entity/services", text="[]")
+        m.get("/api/v1/entity/infrastructure/processes", text=_read_test_file("process_response.json"))
```
Comment (zandrewitte) [inline dynatrace_topology/tests/test_dynatrace_topology.py:70]: Looks like we can extract this part and just pass what is needed each time?

**Review by hruhek — APPROVED** [review]

**Review by zandrewitte — CHANGES_REQUESTED** [review]

File: dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py (hunk 1 above)
Comment (vishalxebia) [reply dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py:299]: ok.

**Review by zandrewitte — APPROVED** [review]

**Pull request merged**